test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp

```diff
@@ -0,0 +1,436 @@
+    eprosima::fastdds::dds::TypeSupport discovery_type(
+            new eprosima::fastdds::statistics::DiscoveryTimePubSubType);
+    eprosima::fastdds::dds::TypeSupport sample_identity_count_type(
+            new eprosima::fastdds::statistics::SampleIdentityCountPubSubType);
+    eprosima::fastdds::dds::TypeSupport physical_data_type(
+            new eprosima::fastdds::statistics::PhysicalDataPubSubType);
+
+    eprosima::fastrtps::types::ReturnCode_t ret = statistics_participant->enable_statistics_datawriter(
+            HISTORY_LATENCY_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(HISTORY_LATENCY_TOPIC));
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(NETWORK_LATENCY_TOPIC));
+    EXPECT_EQ(history_latency_type, statistics_participant->find_type(history_latency_type.get_type_name()));
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            NETWORK_LATENCY_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(NETWORK_LATENCY_TOPIC));
+    EXPECT_EQ(network_latency_type, statistics_participant->find_type(network_latency_type.get_type_name()));
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            PUBLICATION_THROUGHPUT_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(PUBLICATION_THROUGHPUT_TOPIC));
+    EXPECT_EQ(throughput_type, statistics_participant->find_type(throughput_type.get_type_name()));
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            SUBSCRIPTION_THROUGHPUT_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(SUBSCRIPTION_THROUGHPUT_TOPIC));
+    // This topic uses the throughput_type that has been checked previously.
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            RTPS_SENT_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(RTPS_SENT_TOPIC));
+    EXPECT_EQ(rtps_traffic_type, statistics_participant->find_type(rtps_traffic_type.get_type_name()));
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            RTPS_LOST_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(RTPS_LOST_TOPIC));
+    // This topic uses the rtps_traffic_type that has been checked previously.
```

Comment:
It does no harm to check it is still present. Same for all the comments like this one.

Reply:
Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3